[RUBY-3675] Update renewal email to use `TemporaryFirstRenewalReminderService`

jjromeo opened the pull request:
https://eaflood.atlassian.net/browse/RUBY-3675

Add a class `RenewalReminderEmailServiceSwitcher` to switch between the temporary first renewal reminder service and the renewal reminders, controlled by the `enable_renewals` feature flag. Additionally refactor the Temporary reminder emails to more closely resemble the structure of the actual renewal reminder emails.

jjromeo committed:
[RUBY-3675] Update renewal email to use `TemporaryFirstRenewalReminderService`

Modified the `ResendRenewalEmailController` to call `TemporaryFirstRenewalReminderService` instead of `FirstRenewalReminderEmailService` for sending renewal emails, ensuring users are directed to the new registration process.

jjromeo committed:
[RUBY-3675] Update renewal invite email to direct users to new registration

- Renamed `TemporaryFirstRenewalReminderService` to `TemporaryFirstRenewalReminderEmailService` and updated its usage in `resend_renewal_email_controller.rb` and `temporary_first_renewal_reminder_service.rb`.
- Created `temporary_first_renewal_reminder_email_service.rb` to handle sending first renewal reminder emails with a new registration URL.
- Added `temporary_second_renewal_reminder_email_service.rb`

jjromeo committed:
[RUBY-3675] Implement `RenewalServiceSwitcher` for dynamic service selection

- Introduced `RenewalServiceSwitcher` to dynamically select between temporary and non-temporary renewal services based on feature toggles.
- Updated `resend_renewal_email_controller.rb` and `email.rake` tasks to use the new switcher for determining the appropriate service.
- Added `use_temporary_renewal_services` feature toggle in `feature_toggles.yml`.
- Created RSpec tests for `RenewalService

jjromeo committed:
[RUBY-3675] Refactor renewal service switcher to use new feature toggle

Replaced `use_temporary_services?` with `renewals_enabled?` in `RenewalServiceSwitcher` to determine which renewal services to use. Updated the feature toggle in `feature_toggles.yml` from `use_temporary_renewal_services` to `enable_renewals`. Modified corresponding tests in `renewal_service_switcher_spec.rb` to reflect these changes.

jjromeo committed:
Merge branch 'main' into RUBY-3675-wex-update-renewal-invite-email-to-send-user-to-new-registration

brujeo commented on app/services/renewal_service_switcher.rb:
```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+class RenewalServiceSwitcher
```
I think that the service name does not reflect what the service is doing. It's not just switching but also sending out first renewal reminder emails

jjromeo replied:
It is switching between the email services, would `RenewalReminderServiceSwitcher` make it more clear or any other suggestions?

brujeo replied:
maybe FirstRenewalReminderSwitcherService ?

jjromeo replied:
It contains several Renewal reminder services however, not just the First one

jjromeo committed:
Merge branch 'main' into RUBY-3675-wex-update-renewal-invite-email-to-send-user-to-new-registration

jjromeo committed:
[RUBY-3675] Renamed `RenewalServiceSwitcher` to `RenewalReminderEmailServiceSwitcher` for clarity.

- Updated all references to the renamed service in controllers, rake tasks, and specs.
- Adjusted method calls to use the new class name, ensuring consistent naming across the codebase.

jjromeo committed:
Merge branch 'RUBY-3675-wex-update-renewal-invite-email-to-send-user-to-new-registration' of https://github.com/DEFRA/waste-exemptions-back-office into RUBY-3675-wex-update-renewal-invite-email-to-send-user-to-new-registration

jjromeo committed:
[RUBY-3675] Rubocop fix

Refactored `ResendRenewalEmailController` to improve readability by adjusting the line break in the `run` method call of `RenewalReminderEmailServiceSwitcher.first_reminder_email_service`.
